main.tf

```diff
@@ -193,3 +221,29 @@ resource "azurerm_app_service_public_certificate" "functionapp" {
   certificate_location = "CurrentUserMy"
   blob                 = each.value
 }
+
+# Link Private DNS zone to VNet and disable public address to Function App
+# Private DNS Zone should already exist via cpp-terraform-network
+
+data "azurerm_private_dns_zone" "dns_zone" {
+  name                = "privatelink.azurewebsites.azure.net"
+  resource_group_name = var.dns_resource_group_name
+}
+
+resource "azurerm_private_dns_a_record" "dns_record" {
+  #name                = azurerm_function_app.function.name
+  name                = var.asp_os_type == "Linux" ? azurerm_linux_function_app.linux_function.0.name : azurerm_windows_function_app.windows_function.0.name
+  #zone_name           = azurerm_private_dns_zone.rg.name
+  zone_name           = data.azurerm_private_dns_zone.dns_zone.name
+  resource_group_name = var.dns_resource_group_name
+  ttl                 = 300
+  records             = [azurerm_private_endpoint.private_endpoint[0].private_service_connection[0].private_ip_address]
+}
+
+resource "azurerm_private_dns_zone_virtual_network_link" "dns_link" {
+  name                  = "LinkDNSZoneVNet"
```

Comment:
shouldn't this be conditional like privat endpoint based on skus ?

Reply:
also shouldn't this vnet link creation to be done on cpp-terrafrom-networks repo ?